services/cluster-agent/cluster_agent/k8s.py

```
@@ -0,0 +1,138 @@
+        services = _core_v1().list_service_for_all_namespaces()
+    except ApiException as e:
+        if e.status == 403:
+            raise RBACDeniedError(str(e)) from e
+        raise
+    for svc in services.items:
+        if any(sub in svc.metadata.name for sub in name_substrings):
+            return svc.metadata.namespace, svc.metadata.name
+    return None
+
+
+def _get_deployment_sync(namespace: str, name: str):
```

Ruff (0.16.3)

[warning] Missing return type annotation for private function `_get_deployment_sync`

(ANN202)